airflow/providers/apache/druid/hooks/druid.py

```diff
@@ -163,8 +167,8 @@ class DruidDbApiHook(DbApiHook):
 
     conn_name_attr = "druid_broker_conn_id"
     default_conn_name = "druid_broker_default"
-    conn_type = "druid"
-    hook_name = "Druid"
+    conn_type = "druid_broker"
```

Comment:
I assume that this might be a breaking changes. Because if you rename `conn_type` then previously created connection IDs will keep `conn_type=druid` as result Airflow UI have no idea how to deal with this connection type

![image](https://github.com/apache/airflow/assets/3998685/f57eb136-28ff-4bff-9110-006332f630a6)

And fallback to the first available Connection type into the list

![image](https://github.com/apache/airflow/assets/3998685/70590c42-e439-4973-a429-164815f198cf)

So we need to write down information into the change log how to resolve this situation.

Reply:
That makes sense. How about writing a `alembic` migration for that?

Reply:
We do not have mechanism to run alembic when upgrading provider version.

I think noting it the changelog and making it a major release with breaking change should be enough.
cc @potiuk WDYT?

Reply:
Or adding backwards compatibility to handle both cases - I think it should be possible.

Reply:
The problem that `DruidDbApiHook` use `DbApiHook` and this might be the single place outside of the UI where `conn_type` also matter. And it could be hard, but not impossible to make it by backward compatible. Better just keep `conn_type="druid"` since it required almost zero effort.

Reply:
Indeed the easiest way would be switch to `druid`. how about also adding a `warning` that the name would be change in the next major release?

Reply:
We can decide on deprecating first or on cutting major release immediately.
We always prefer to deprecate first if we can.